### PR TITLE
Make it possible to install multiple kernel specs

### PR DESCRIPTION
**Pull request by JanSchulz**
Closes: https://github.com/IRkernel/IRkernel/issues/157

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,11 +15,26 @@ For detailed requirements and install instructions see [irkernel.github.io](http
...
+install.packages(c(repr','IRkernel','IRdisplay'),
                  repos = c('http://irkernel.github.io/', getOption('repos')))
 IRkernel::installspec()
 ```
 
+Per default `IRkernel::installspec()` will install a kernel with the name `"ir"` and a
+display name of `"R"`. Multiple calls will overwrite the kernel with a kernel spec pointing to the last
+R interpreter you called that commands from. You can install kernels for multiple versions of R
+by supplying a `name` and `displayname` argument to the `installspec()` call (You still need to
+install these packages in all interpreter you want to run as a jupyter kernel!):
```
Comment (takluyver) [inline README.md]: I think this description could be made a bit shorter, but I'm happy to have a go at it after we merge.

**Comment by takluyver** [comment]
I think this looks good; I'll give @flying-sheep a chance to look at it before we hit the merge button.

File: README.md [hunk 2]
```diff
@@ -15,11 +15,26 @@ For detailed requirements and install instructions see [irkernel.github.io](http
 We provide Windows and Mac OS X binary packages of all the needed packages:
 
 ```r
-install.packages(c('rzmq','repr','IRkernel','IRdisplay'),
+install.packages(c(repr','IRkernel','IRdisplay'),
```
Comment (JanSchulz) [inline README.md]: I do hope that's correct: now that we use a zmq package from cran...

File: README.md [hunk 3]
```diff
@@ -33,7 +48,8 @@ If you have Jupyter installed, you can create a notebook using IRkernel from the
 You can also start other interfaces with an R kernel:
 
 ```bash
-# “ir” is the kernel name installed by the above 'IRkernel::installspec()'
```
Comment (JanSchulz) [inline README.md]: R complained about "non ascii" during the build... Also, git diff didn't like them...

File: README.md (hunk 2 above)
Comment (takluyver) [inline README.md]: I guess it will be once we do a new release. It's not quite clear whether README should refer to the development version or the released version. But I guess we should do a new release soon (I feel like I'm always saying that ;-).

File: R/installspec.r [hunk 4]
```diff
@@ -14,11 +20,11 @@ installspec <- function(user = TRUE) {
...
         }
     }
-    
+
     if (!found_binary)
-        stop('Jupyter or IPython 3.0 has to be installed but could neither run “jupyter” nor “ipython”, “ipython2” or “ipython3”.
-             (Note that “ipython2” is just IPython for Python 2, but still may be IPython 3.0)')
-    
+        stop('Jupyter or IPython 3.0 has to be installed but could neither run "jupyter" nor "ipython", "ipython2" or "ipython3".
+             (Note that "ipython2" is just IPython for Python 2, but still may be IPython 3.0)')
```
Comment (flying-sheep) [inline R/installspec.r]: please use `dQuote()` instead of straight quotes

File: README.md (hunk 2 above)
Comment (flying-sheep) [inline README.md]: you deleted a single quote too many, this is now invalid syntax

File: README.md (hunk 1 above)
Comment (flying-sheep) [inline README.md]: > in all interpreter**s**

File: README.md [hunk 5]
```diff
@@ -15,11 +15,26 @@ For detailed requirements and install instructions see [irkernel.github.io](http
...
+display name of `"R"`. Multiple calls will overwrite the kernel with a kernel spec pointing to the last
+R interpreter you called that commands from. You can install kernels for multiple versions of R
+by supplying a `name` and `displayname` argument to the `installspec()` call (You still need to
+install these packages in all interpreter you want to run as a jupyter kernel!):
+
+```r
+# in R 3.2.2
+IRkernel::installspec(name="ir322", displayname="R 3.2.2")
+# in R 3.1
+IRkernel::installspec(name="ir31", displayname="R 3.1")
```
Comment (flying-sheep) [inline README.md]: again: spaces around `=`, please

File: README.md (hunk 3 above)
Comment (flying-sheep) [inline README.md]: please undo this change. there’s no need to remove them from .md files

File: test_ir.py [hunk 6]
```diff
@@ -18,5 +18,14 @@ class IRkernelTests(jkt.KernelTests):
...
 
+class InstallspecTests(jkt.KernelTests):
+    # just a small test, it's the same kernel after all...
+    kernel_name = "testir"
+
+    language_name = "R"
+
+    code_hello_world = "print('hello, world')"
+
+
```
Comment (flying-sheep) [inline test_ir.py]: please only use double quotes around strings that contain single quotes.

**Comment by flying-sheep** [comment]
added some pedantic codeing style stuff that should be fixed, but otherwise it looks good!

File: R/installspec.r (hunk 4 above)
Comment (JanSchulz) [inline R/installspec.r]: ok, I'm passionate about the smart code thing (or using any non-unicode char) in code . Or better: violently against it: it's messing with windows cmd output (such as git diff) and it's breaking when I c&p it into an example call. Smart quotes should be outlawed in any source code files and even in published documentation...

R seems to agree, at least it issues a warning during my builds...

Anyway: I will revert these parts as the have nothing to do with the original issue...

File: README.md (hunk 5 above)
Comment (JanSchulz) [inline README.md]: guess I'm also to python centric... No spaces around "="  in arguments to a function call...

File: test_ir.py (hunk 6 above)
Comment (JanSchulz) [inline test_ir.py]: thats copied from above, where double quotes are used as well. Seems that this files has mixed quotes all over?

**Comment by JanSchulz** [comment]
Repushed with most of the pedantic code styles annotations fixed (under protest in the case of the smart quotes...grummel) :-) I left out the python file, as this is mixing quotes all over...

File: man/installspec.Rd [hunk 7]
```diff
@@ -4,12 +4,21 @@
 \alias{installspec}
 \title{Install the kernelspec to tell Jupyter (or IPython ≥ 3) about IRkernel.}
 \usage{
-installspec(user = TRUE)
+installspec(user = TRUE, name = "ir", displayname = "R")
```
Comment (JanSchulz) [inline man/installspec.Rd]: I couldn't get `devtools::document()` to change this to single quotes...

http://r-pkgs.had.co.nz/man.html

File: test_ir.py (hunk 6 above)
Comment (takluyver) [inline test_ir.py]: I wrote it, so that's expected. I'm completely inconsistent with quotes in code. The only time I pay attention to them is when I want to put one or other kind of quote inside the string itself. I also don't particularly care about making them consistent, so don't feel we need to 'clean up' the Python code.

File: R/installspec.r (hunk 4 above)
Comment (flying-sheep) [inline R/installspec.r]: no, i didn’t mean that you should revert that. i meant you should do

```
paste(..., '... neither run', dQuote('jupyter'), 'nor', dQuote('ipython'), ',', ...)`

File: test_ir.py (hunk 6 above)
Comment (flying-sheep) [inline test_ir.py]: i’ll do it anyway then afterwards.

File: man/installspec.Rd (hunk 7 above)
Comment (flying-sheep) [inline man/installspec.Rd]: that’s ok, that’s just R deparsing stuff i think.

File: R/installspec.r (hunk 4 above)
Comment (takluyver) [inline R/installspec.r]: -1 to this. It's making the code uglier and more complex for no good reason. Programmers are used to seeing plain ascii quotes anyway.

File: test_ir.py (hunk 6 above)
Comment (takluyver) [inline test_ir.py]: I'd rather you didn't, to be honest. Doing things like that breaks tools like git blame, and it really doesn't help anyone.

File: R/installspec.r (hunk 4 above)
Comment (flying-sheep) [inline R/installspec.r]: oh, damn, too late. sorry.

but i [don’t think it’s that bad](https://github.com/IRkernel/IRkernel/blob/bcaea6800a76c76740d6f01818d302674aa73da6/R/installspec.r#L25-L27).

fucking R/Windows not getting their collective shit together

Comment (flying-sheep) [inline R/installspec.r]: oh and plain ascii quotes mean “string” to me as a programmer.

obviously i can interpret them contextually but i prefer to see `"string-literals"` marked up in monospace and with plain quotes, and “something quoted” in typographical quotes.

File: test_ir.py (hunk 6 above)
Comment (flying-sheep) [inline test_ir.py]: good incentive to start being consistent in the first place :stuck_out_tongue_winking_eye: 

and in this case we only have like 20 lines in there yet anyway.

also please remember when i re-did *all* the IRkernel code style. i’m sure you like working on the codebase more now that it doesn’t look like it did before.

Comment (takluyver) [inline test_ir.py]: > i’m sure you like working on the codebase more now that it doesn’t look like it did before.

Not especially. ;-) It's not worse, I just don't really care about style. Possibly this comes from working on lots of different projects, and on some quite old codebases - I'm used to looking at code in all sorts of styles, and the details of that don't really bother me.

The thing that annoys me is when projects have automatic style checkers and won't accept working, readable PRs until the code complies with every last niggling detail of the style guide.

Comment (flying-sheep) [inline test_ir.py]: yeah, that’s stupid. but i do accept tests failing because of it (at the end, so that you can see that everything else works)
